cmd/src/batch_common.go

```diff
@@ -477,7 +479,13 @@ func parseBatchSpec(file string, svc *service.Service) (*batcheslib.BatchSpec, s
 		return nil, "", errors.Wrap(err, "reading batch spec")
 	}
 
-	spec, err := svc.ParseBatchSpec(data)
+	p, err := filepath.Abs(file)
```

Comment:
We'll need to special case `file == "-"` and `file == ""`, per the code in `batchOpenFileFlag` to handle reading from stdin. (In that case, `dir` is presumably going to be the current working directory.)